src/crimson/os/seastore/cached_extent.h

```diff
@@ -398,6 +398,7 @@ class CachedExtent : public boost::intrusive_ref_counter<
     if (!addr.is_relative()) {
       return addr;
     } else if (is_mutation_pending()) {
+      ceph_assert(addr.is_record_relative());
```

Comment:
probably we could replace some of ceph_assert() calls with plain assert() in critical paths once the function of code is verified and stablized a little bit.